### PR TITLE
REF: delegate more IntervalIndex methods

### PR DESCRIPTION
**Pull request by jbrockmendel**
cc @jschendel is the usage of _simple_new instead of _shallow_copy `take` important?  Should it always be the case that `idx.close == idx._data.closed`?

### CODE REVIEW

**Review by jreback — APPROVED** [review]
lgtm. @jschendel

**Comment by jschendel** [comment]
> is the usage of `_simple_new` instead of `_shallow_copy` in `take` important?

I don't think so.

> should it always be the case that `idx.closed == idx._data.closed`?

Yes.

**Review by jschendel — APPROVED** [review]

**Comment by jschendel** [comment]
thanks @jbrockmendel

File: pandas/core/indexes/interval.py [hunk 1]
```diff
@@ -393,25 +400,6 @@ def __contains__(self, key) -> bool:
...
-        Examples
-        --------
-        >>> idx = pd.IntervalIndex.from_arrays([0, np.nan, 2], [1, np.nan, 3])
-        >>> idx.to_tuples()
-        Index([(0.0, 1.0), (nan, nan), (2.0, 3.0)], dtype='object')
-        >>> idx.to_tuples(na_tuple=False)
-        Index([(0.0, 1.0), nan, (2.0, 3.0)], dtype='object')
-        """,
-        )
-    )
```
Comment (jorisvandenbossche) [inline pandas/core/indexes/interval.py]: By removing this, the docstring of `IntervalIndex.to_tuples` is now no longer correct (and misses the example).

The delegation mechanism will need a way to change docstrings?

**Pull request merged**